java/java.source.base/src/org/netbeans/modules/java/source/parsing/FileObjects.java

```diff
@@ -949,6 +949,34 @@ static Pair<String,List<URL>> parseModulePatches(@NonNull final Iterator<? exten
         return null;
     }
 
+    public static URI getZipPathURI(URI zipURI, String resourceName) {
+        try {
+            //Optimistic try and see
+            return new URI ("jar:"+zipURI.toString()+"!/"+resourceName); //NOI18N
+        } catch (URISyntaxException e) {
+            //Need to encode the resName part (slower)
+            final StringBuilder sb = new StringBuilder ();
+            final String[] elements = resourceName.split("/");           //NOI18N
+            try {
+                for (int i = 0; i< elements.length; i++) {
+                    String element = elements[i];
+                    element = URLEncoder.encode(element, "UTF-8");       //NOI18N
+                    element = element.replace("+", "%20");               //NOI18N
```

Comment:
Replacing "+" with URL-encoded `space` ?

Reply:
`URLEncoder.encode` apparently replaces spaces with `+`. So this merely undoes that.

Reply:
Because `URLEncoder` is not designed for encoding URLs. Are you sure that's the only possible problematic character?  There are libraries in NetBeans already that could properly encode the path if not.

Reply:
Seems space is the only character:
https://github.com/openjdk/jdk/blob/670b4567cf8229c9fd40c639a04dd1f1b7cfd551/src/java.base/share/classes/java/net/URLEncoder.java#L233

Please note this code is merely moved from one place to another, verbatim. I am not really confident to change that code, unless there's a bug in it that needs fixing.

Reply:
OK, makes sense.  I think other than space, it might just be escaping more than it needs to in this context.